### PR TITLE
change OS X to macOS

### PR DESCRIPTION
**Pull request by tdm00**
##### SUMMARY
Replace all references to Mac OS X with macOS
Apple shortened the name to "macOS" in 2016

##### ISSUE TYPE
 - Docs Pull Request

### CODE REVIEW

**Review by mkrizek — COMMENTED** [review]
Maybe we could fix all occurrences of `OS X` in the docs? Also, looks like there are several uses of `macOS/OS X` already.

**Comment by samdoran** [comment]
I'm 👍 to changing this in all the docs. While the name is similar to Classic Mac OS, it is the new proper name for the operating system. Plus, Ansible doesn't work on classic Mac OS, so there should be no ambiguity in the context of Ansible which version of the Macintosh operating system we're referring to.

**Comment by tdm00** [comment]
@mkrizek and @samdoran I've updated all of the occurrences that I could find from Mac OS X to macOS. I also updated the pull request description based on the provided template. Sorry, I did this pull while pretty tired and somehow glossed over that.

File: docs/docsite/rst/scenario_guides/guide_gce.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ The GCE modules all require the apache-libcloud module which you can install fro
 
     $ pip install apache-libcloud
 
-.. note:: If you're using Ansible on Mac OS X, libcloud also needs to access a CA cert chain. You'll need to download one (you can get one for `here <http://curl.haxx.se/docs/caextract.html>`_.)
+.. note:: If you're using Ansible on Mac macOS, libcloud also needs to access a CA cert chain. You'll need to download one (you can get one for `here <http://curl.haxx.se/docs/caextract.html>`_.)
```
Comment (samdoran) [inline docs/docsite/rst/scenario_guides/guide_gce.rst]: Drop the extra "Mac" here

File: lib/ansible/module_utils/urls.py [hunk 2]
```diff
@@ -625,7 +625,7 @@ def get_ca_certs(self):
         to_add_fd, to_add_path = tempfile.mkstemp()
         to_add = False
 
-        # Write the dummy ca cert if we are running on Mac OS X
+        # Write the dummy ca cert if we are running on Mac macOS
```
Comment (samdoran) [inline lib/ansible/module_utils/urls.py]: Drop the extra "Mac" here

File: lib/ansible/plugins/filter/core.py [hunk 3]
```diff
@@ -275,7 +275,7 @@ def get_encrypted_password(password, hashtype='sha512', salt=None):
 
         if not HAS_PASSLIB:
             if sys.platform.startswith('darwin'):
-                raise AnsibleFilterError('|password_hash requires the passlib python module to generate password hashes on Mac OS X/Darwin')
+                raise AnsibleFilterError('|password_hash requires the passlib python module to generate password hashes on Mac macOS/Darwin')
```
Comment (samdoran) [inline lib/ansible/plugins/filter/core.py]: Drop the extra "Mac" here.

File: test/units/modules/network/f5/fixtures/load_ltm_irules.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
...
... {\n           log -noname accesscontrol.local1.debug \"$static::ACCESS_LOG_PREFIX Setting $user_key => $user_key_value $inactivity_timeout $max_sess_timeout in table $static::ACCESS_USERKEY_TBLNAME\"\n\n           table set -subtable  $static::ACCESS_USERKEY_TBLNAME $user_key $user_key_value $inactivity_timeout $max_sess_timeout\n        } else {\n           log -noname accesscontrol.local1.debug \"$static::ACCESS_LOG_PREFIX Deleting $user_key in table $static::ACCESS_USERKEY_TBLNAME\"\n\n           table delete -subtable  $static::ACCESS_USERKEY_TBLNAME $user_key\n        }\n\n        if { $f_delete_session == 1 } {\n           ACCESS::session remove\n           set f_delete_session 0\n           log -noname accesscontrol.local1.debug \"$static::ACCESS_LOG_PREFIX Removing the session for $user_key.\"\n        }\n    }\ndefinition-signature CZnUb3niz9wZPWvOmjDB0Dy4ixqjBEhIZrAVGt8VYe7+wZkhcBUFTADz3S1y5uomVwhRkGL20PLH7tfanDlpr3+IppgAGQlp98sPUl5ndEoWA4Rr90QiRGNRl/V7jWK58SOdJCQOirnutVMoeYjBWLwuprXGts08PO0WML5s0xJNOY7WPuGNeG+7Ht2pIB0vu80CgnCNGZJGZH0QR3kMVOx3yUN0ro5bAOmQ/XWel4qkj0F5DN9ufvsmKtTvb+Lc3y+5PHGbbFAQIrZ7lntZUJl/F8e/d26HE3spmZzQpPzi16qYWaMOxbvT6oedxpyhwbmJLiRNGyZmnT6kHj93FA==",
```
Comment (samdoran) [inline test/units/modules/network/f5/fixtures/load_ltm_irules.json]: Drop the extra "Mac" here.

File: test/units/modules/network/f5/fixtures/load_ltm_irules.json [hunk 5]
```diff
@@ -42,7 +42,7 @@
...
... {\n           log -noname accesscontrol.local1.debug \"$static::ACCESS_LOG_PREFIX Setting $user_key => $user_key_value $inactivity_timeout $max_sess_timeout in table $static::ACCESS_USERKEY_TBLNAME\"\n\n           table set -subtable  $static::ACCESS_USERKEY_TBLNAME $user_key $user_key_value $inactivity_timeout $max_sess_timeout\n        } else {\n           log -noname accesscontrol.local1.debug \"$static::ACCESS_LOG_PREFIX Deleting $user_key in table $static::ACCESS_USERKEY_TBLNAME\"\n\n           table delete -subtable  $static::ACCESS_USERKEY_TBLNAME $user_key\n        }\n\n        if { $f_delete_session == 1 } {\n           ACCESS::session remove\n           set f_delete_session 0\n           log -noname accesscontrol.local1.debug \"$static::ACCESS_LOG_PREFIX Removing the session for $user_key.\"\n        }\n    }\ndefinition-signature feX9LM+vB6YOEdVF+EA1JtNyVkPaB7gwdW0JzaB083MXl4yPP2nZnjm+WAx3YQhsmLttq5UkPl1zHpr5H9cwJX1bu9BNMi/+n0bIqWOipDHhhSYQ+TH+a5jQUSeftISr52BSQxh0cQKZkzM3rFU/qRZn9D9Dbf0kDGiDC1KWwVosrdjp5tVHOiQXWx8zybbGPFfgBcIBE6IvOvGbh5ohebVL2ADZm0URRj2NM4ZvZ2T3C14k2rHGXnDdRsvhmf5USZ+FH1hoKtWRxqtFjkWIaqw8leenXeot1j2bdKy92/AVTC9oZj1HJN1ePuQo5v414zlUhMEtkVy/gaxvj1+vPQ==",
```
Comment (samdoran) [inline test/units/modules/network/f5/fixtures/load_ltm_irules.json]: Drop the extra "Mac" here.

**Review by samdoran — CHANGES_REQUESTED** [review]

**Comment by tdm00** [comment]
Performed a rebase of the latest changes in devel to this pull request to reduce the possibility of any conflicts.

**Comment by acozine** [comment]
Thanks @tdm00 for updating the docs to use the correct name for Apple's operating system. I appreciate your attention to detail, your response to @samdoran's requests for changes, and your contribution. 

Looking at this PR, though, I see a lot of unrelated commits, including changes to non-docs files like `lib/ansible/module_utils/network/eos/eos.py`. The build is also failing. I suspect something went wrong when you rebased the branch. 

I'd like to get the OS X -> macOS change merged from a clean PR (one that only includes the docs changes and passes our CI suite). If you don't want to (or aren't sure how to) get those changes onto a clean branch/PR, let me know - I can coach you through it or just take care of it.

**Comment by tdm00** [comment]
@acozine thank you. Please correct me if my understanding below is incorrect:

1. Close this pull request
2. Go to my repo and patch-1 branch and create a new branch from commit c60038d, which was done prior to the rebase.
3. Push the new branch to my GitHub repo
4. Open a new Pull Request

**Comment by samdoran** [comment]
@tdm00 This looks like a rebase gone wrong. You should not need to open a new PR. You can try rebasing again, then force-push to the same branch to overwrite the current history.

**Comment by acozine** [comment]
@tdm00 the most important part is creating a branch that contains only the OS X -> macOS changes - to do that you can try re-re-basing the branch as @samdoran suggests. If that doesn't work, you can try this (commands below assume the main ansible/ansible repo is referred to as `origin` on your system - you can check this by doing `git remote -v`):
1. fetch the latest version of `devel` branch: `git fetch origin`
2. create a fresh local branch off `devel` branch: `git checkout -b fresh_branch origin/devel`
3. cherry-pick the commits you want for this PR to the fresh local branch (I think that would be: 
e76f2d1e638170872159e98dc112ac43ec013f22, c1d60ddb0dced40c70eefef6ccae5adb15290d95, 85b3cd4e20559bb58de2f7544fa55fd09f2cc4e7, 8a58efa7131ac0fe99b6af109c749b4c4d18cf1e, 13ca452619d8f52af1b662bea23ba97bf6bef8d6, and c60038d5943435218eb3f914b98c161841507461): `git cherry-pick ####` for each commit hash
4. rename your `patch-1` branch: `git branch -m patch-1 patch-1-bad-rebase`
5. force-push your clean branch to this pr: `git push -f your_remote fresh_branch:patch-1`

**Comment by tdm00** [comment]
@acozine and @samdoran I did a `git reset --hard to c60038d` and then did a force push `git push -f origin patch-1`. This should give you a branch that has just the documentation changes.

**Review by samdoran — APPROVED** [review]

**Comment by samdoran** [comment]
Looks good. Thanks!

**Pull request merged**